lektor/admin/static/js/widgets/multiWidgets.jsx

```diff
@@ -1,126 +1,98 @@
 'use strict'
 
 import React from 'react'
-import createReactClass from 'create-react-class'
 import { flipSetValue } from '../utils'
 import i18n from '../i18n'
-import { BasicWidgetMixin } from './mixins'
+import { getInputClass, widgetPropTypes } from './mixins'
 
-const CheckboxesInputWidget = createReactClass({
-  displayName: 'CheckboxesInputWidget',
-  mixins: [BasicWidgetMixin],
-
-  statics: {
-    deserializeValue: (value) => {
-      if (value === '') {
-        return null
-      }
-      let rv = value.split(',').map((x) => {
-        return x.match(/^\s*(.*?)\s*$/)[1]
-      })
-      if (rv.length === 1 && rv[0] === '') {
-        rv = []
-      }
-      return rv
-    },
-
-    serializeValue: (value) => {
-      return (value || '').join(', ')
-    }
-  },
+function deserializeCheckboxValue (value) {
+  if (value === '') {
+    return null
+  }
+  let rv = value.split(',').map((x) => {
+    return x.match(/^\s*(.*?)\s*$/)[1]
+  })
+  if (rv.length === 1 && rv[0] === '') {
+    rv = []
+  }
+  return rv
+}
 
-  onChange: function (field, event) {
-    const newValue = flipSetValue(this.props.value,
-      field, event.target.checked)
-    if (this.props.onChange) {
-      this.props.onChange(newValue)
-    }
-  },
+function serializeCheckboxValue (value) {
+  return (value || '').join(', ')
+}
 
-  isActive: function (field) {
-    let value = this.props.value
+function checkboxIsActive (field, props) {
+  let value = props.value
+  if (value == null) {
+    value = props.placeholder
     if (value == null) {
-      value = this.props.placeholder
-      if (value == null) {
-        return false
-      }
+      return false
     }
-    for (const item of value) {
-      if (item === field) {
-        return true
-      }
+  }
+  for (const item of value) {
+    if (item === field) {
+      return true
     }
-    return false
-  },
+  }
+  return false
+}
 
-  render () {
-    let { className, value, placeholder, type, ...otherProps } = this.props // eslint-disable-line no-unused-vars
-    className = (className || '') + ' checkbox'
+export function CheckboxesInputWidget (props) {
```

Comment:
Static methods wouldn't work, since we lookup the (de)serializeValue function on the class instance. Also the function component let's me avoid the assignment of a bound event handler in the constructor. Two points for a function component IMHO.

Reply:
Scratch that, the part about the static methods is nonsense...

Reply:
Done